kani-compiler/src/kani_middle/transform/loop_contracts.rs

```diff
@@ -114,7 +114,7 @@ impl TransformPass for LoopContractPass {
                     let mut loop_queue: VecDeque<BasicBlockIdx> = VecDeque::new();
                     queue.push_back(0);
 
-                    while let Some(bb_idx) = queue.pop_front().or(loop_queue.pop_front()) {
+                    while let Some(bb_idx) = queue.pop_front().or_else(|| loop_queue.pop_front()) {
```

Comment:
Is this change necessary?

Reply:
Yes, with `or` the `loop_queue` will also `pop_front` at each iteration, while we want it to `pop_front` only when `queue` is empty.